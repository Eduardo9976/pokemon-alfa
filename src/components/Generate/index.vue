<template>
  <li>
    <img :src="`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`" :alt="name" />
    <div class="infos">
      <p>{{ name }}</p>
      <p>#{{ id }}</p>
    </div>
  </li>
</template>

<script>

export default {
  props: {
    name: { type: String, required: true },
    id: { type: [String, Number], required: true }
  }
}
</script>

<style scoped>
li {
  background: #ffffff;
  width: 200px;
  border-radius: 4px;
  margin: 5px;
  padding: 5px;
  text-transform: capitalize;
}
img {
  width: 96px;
  height: 96px;
  margin: 0 auto;
  display: block;
}
.infos {
  display: flex;
  justify-content: space-between;
  padding: 10px;
}
</style>
