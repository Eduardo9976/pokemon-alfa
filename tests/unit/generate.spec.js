import Generate from '../../src/components/Generate'
import { shallowMount } from '@vue/test-utils'

describe('<Generate />', () => {
    it('should render correctly component', () => {
        const wrapper = shallowMount(Generate, {
            props: { name: 'charizard', id: '6' }
        })
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.find('li').exists()).toBe(true)
        expect(wrapper.find('img').exists()).toBe(true)
        expect(wrapper.find('.infos').exists()).toBe(true)
        expect(wrapper.find('p').exists()).toBe(true)
    })
    it('should render correctly the components itens', () => {
        const wrapper = shallowMount(Generate, {
            props: { name: 'charizard', id: '6' }
        })
        expect(wrapper.find('p').text()).toBe('charizard')
        expect(wrapper.find('p:nth-child(2)').text()).toBe('#6')
        expect(wrapper.find('img').attributes('alt')).toBe('charizard')
        expect(wrapper.find('img').attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png')
    })
})