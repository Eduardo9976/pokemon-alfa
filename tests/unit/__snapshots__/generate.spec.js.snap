// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Generate /> should render correctly component 1`] = `
<li><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png" alt="charizard">
  <div class="infos">
    <p>charizard</p>
    <p>#6</p>
  </div>
</li>
`;
